sensing/pointcloud_preprocessor/src/outlier_filter/ring_outlier_filter_nodelet.cpp

```diff
@@ -218,52 +214,47 @@ void RingOutlierFilterComponent::faster_filter(
           output_ptr->y = p[1];
           output_ptr->z = p[2];
         } else {
-          *output_ptr = *input_ptr;
+          output_ptr->x = input_ptr->x;
+          output_ptr->y = input_ptr->y;
+          output_ptr->z = input_ptr->z;
         }
         const float & intensity =
-          *reinterpret_cast<const float *>(&input->data[indices[i] + intensity_offset]);
+          *reinterpret_cast<const float *>(&input->data[indices[i] + input_intensity_offset]);
         output_ptr->intensity = intensity;
 
+        const std::uint8_t & return_type = *reinterpret_cast<const std::uint8_t *>(
+          &input->data[indices[i] + input_return_type_offset]);
+        output_ptr->return_type = return_type;
+
+        const std::uint8_t & channel =
+          *reinterpret_cast<const std::uint8_t *>(&input->data[indices[i] + input_channel_offset]);
+        output_ptr->channel = channel;
+
         output_size += output.point_step;
       }
     } else if (publish_outlier_pointcloud_) {
       for (int i = walk_first_idx; i < walk_last_idx; i++) {
-        PointXYZIRADRT outlier_point;
-
-        auto input_ptr = reinterpret_cast<const PointXYZIRADRT *>(&input->data[indices[i]]);
+        auto input_ptr = reinterpret_cast<const InputPointType *>(&input->data[indices[i]]);
+        InputPointType outlier_point = *input_ptr;
         if (transform_info.need_transform) {
           Eigen::Vector4f p(input_ptr->x, input_ptr->y, input_ptr->z, 1);
           p = transform_info.eigen_transform * p;
           outlier_point.x = p[0];
           outlier_point.y = p[1];
           outlier_point.z = p[2];
-        } else {
-          outlier_point = *input_ptr;
         }
-        outlier_point.intensity =
-          *reinterpret_cast<const float *>(&input->data[indices[i] + intensity_offset]);
-        outlier_point.ring =
-          *reinterpret_cast<const uint16_t *>(&input->data[indices[i] + ring_offset]);
-        outlier_point.azimuth =
-          *reinterpret_cast<const float *>(&input->data[indices[i] + azimuth_offset]);
-        outlier_point.distance =
-          *reinterpret_cast<const float *>(&input->data[indices[i] + distance_offset]);
-        outlier_point.return_type =
-          *reinterpret_cast<const uint8_t *>(&input->data[indices[i] + return_type_offset]);
-        outlier_point.time_stamp =
-          *reinterpret_cast<const float *>(&input->data[indices[i] + time_stamp_offset]);
+
         outlier_pcl->push_back(outlier_point);
       }
     }
   }
 
-  setUpPointCloudFormat(input, output, output_size, /*num_fields=*/4);
+  setUpPointCloudFormat(input, output, output_size);
 
   if (publish_outlier_pointcloud_) {
     PointCloud2 outlier;
     pcl::toROSMsg(*outlier_pcl, outlier);
     outlier.header = input->header;
-    outlier_pointcloud_publisher_->publish(outlier);
```

Comment:
It looks like this publish is deleted without a replacement. Please check.

Reply:
Thanks, dunno how this came to be.
Addressed in [6793245](https://github.com/autowarefoundation/autoware.universe/pull/6996/commits/6793245f20a97f874eb90e448ae51edaa38ddc82)

Reply:
Confirmed working :+1: